src/compiler/checker.ts

```diff
@@ -19112,7 +19110,7 @@ namespace ts {
                         }
                     }
 
-                    const isPerformingCommonPropertyChecks = (relation !== comparableRelation || !(source.flags & TypeFlags.Union) && isLiteralType(source)) &&
+                    const isPerformingCommonPropertyChecks = (relation !== comparableRelation || isUnitType(source)) &&

```

Comment:
I believe the problem with `isUnitType` is that it doesn't consider enum literal types.

https://github.com/microsoft/TypeScript/pull/50423/files#r954112029

So we won't catch assignability from an enum literal to a weak type with no overlap.

```ts
enum E { A = "A" }

// these will differ
let x: { nope?: any } = E.A;
let y: { nope?: any } = "A";
```

Reply:
Enum literal types are definitely considered unit types. It's a bit subtle, `isUnitType` checks for `TypeFlags.Unit`, which includes `TypeFlags.Literal`, which includes `TypeFlags.StringLiteral` and `TypeFlags.NumberLiteral`. For enum literal types, the `TypeFlags.EnumLiteral` flag is always combined with one of the latter two, so `isUnitType` will always be `true`.

Reply:
Just to be sure, I verified both of your examples generate errors.

Reply:
Great! Can you add that as a test case?